core/src/main/scala/org/apache/spark/scheduler/cluster/StandaloneSchedulerBackend.scala

```diff
@@ -111,15 +112,19 @@ private[spark] class StandaloneSchedulerBackend(
     // ExecutorAllocationManager will send the real initial limit to the Master later.
     val initialExecutorLimit =
       if (Utils.isDynamicAllocationEnabled(conf)) {
+        if (coresPerExecutor.isEmpty) {
+          logWarning("Dynamic allocation without explicitly setting spark.executor.cores " +
+            "detected, you may get more executors allocated than expected. It's recommended to " +
+            "set spark.executor.cores explicitly. Check this issue for more details: " +
+            "https://issues.apache.org/jira/browse/SPARK-30299")
```

Comment:
nit: "Please check [SPARK-30299](https://issues.apache.org/jira/browse/SPARK-30299) for more details."